src/routes/teams.rs

```diff
@@ -1,25 +1,227 @@
+            .await
+            .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+        if team_member.is_some() {
+            let team_members: Vec<crate::models::teams::TeamMember> = members_data
+                .into_iter()
+                .map(|data| crate::models::teams::TeamMember {
+                    user_id: data.user_id.into(),
+                    name: data.name,
+                    role: data.role,
+                    permissions: data.permissions,
+                })
+                .collect();
+
+            return Ok(HttpResponse::Ok().json(team_members));
+        }
+    }
+
     let team_members: Vec<crate::models::teams::TeamMember> = members_data
         .into_iter()
         .map(|data| crate::models::teams::TeamMember {
             user_id: data.user_id.into(),
             name: data.name,
             role: data.role,
+            permissions: Permissions::default(),
         })
         .collect();
 
     Ok(HttpResponse::Ok().json(team_members))
 }
+
+#[post("{id}/join")]
+pub async fn join_team(
+    req: HttpRequest,
+    info: web::Path<(TeamId,)>,
+    pool: web::Data<PgPool>,
+) -> Result<HttpResponse, ApiError> {
+    let team_id = info.into_inner().0.into();
+    let current_user = get_user_from_headers(req.headers(), &**pool)
+        .await
+        .map_err(|_| ApiError::AuthenticationError)?;
+
+    // Edit Team Member to set Accepted to True
+    TeamMember::edit_team_member(
+        team_id,
+        current_user.id.into(),
+        None,
+        None,
+        Some(true),
+        &**pool,
+    )
+    .await?;
+
+    Ok(HttpResponse::Ok().body(""))
+}
+
+#[post("{id}/members")]
+pub async fn add_team_member(
+    req: HttpRequest,
+    info: web::Path<(TeamId,)>,
+    pool: web::Data<PgPool>,
+    new_member: web::Json<crate::models::teams::TeamMember>,
+) -> Result<HttpResponse, ApiError> {
+    let team_id = info.into_inner().0.into();
+
+    let mut transaction = pool
+        .begin()
+        .await
+        .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+    let current_user = get_user_from_headers(req.headers(), &**pool)
+        .await
+        .map_err(|_| ApiError::AuthenticationError)?;
+    let team_member =
+        TeamMember::get_from_user_id(team_id, current_user.id.into(), &**pool).await?;
+
+    if let Some(member) = team_member {
+        if member.permissions.contains(Permissions::MANAGE_INVITES)
+            && new_member.role != crate::models::teams::OWNER_ROLE
+        {
+            if !member.permissions.contains(new_member.permissions) {
+                return Err(ApiError::AuthenticationError);
+            }
+
+            let new_id =
+                crate::database::models::ids::generate_team_member_id(&mut transaction).await?;
+            TeamMember {
+                id: new_id,
+                team_id,
+                user_id: new_member.user_id.into(),
+                name: new_member.name.clone(),
```

Comment:
I don't think that it's a good idea to have the inviting user be able to specify how the new member's username will appear... This should just use the other member's username as a default, and allow them to change it later.